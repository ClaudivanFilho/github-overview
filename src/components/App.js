import React, { Component } from 'react'
import { Link, Route } from 'react-router-dom'
import { ToastContainer, toast } from 'react-toastify'
import 'react-toastify/dist/ReactToastify.css'

import RepositoriesPage from './RepositoriesPage'
import HomePage from './HomePage'
import RepositoryContext from '../contexts/RepositoryContext'
import ToastContext from '../contexts/ToastContext'
import Api from '../api'

export default class App extends Component {
  state = {
    repositories: [],
    originalRepositories: [],
  }

  componentDidMount() {
    this.refetchRepositories()
  }

  updateRepositories = (reps, originalReps) => {
    this.setState({
      repositories: reps,
      originalRepositories: originalReps || this.state.originalRepositories,
    })
  }

  refetchRepositories = () => {
    Api.fetchRepositories()
      .then(res => this.setState({ originalRepositories: res, repositories: res }))
  }

  openToast = (message, isError) => {
    const toastConfig = {
      position: toast.POSITION.TOP_CENTER,
      autoClose: isError ? 4000 : 2000,
      hideProgressBar: true,
      closeOnClick: true,
      draggable: false,
    }
    if (isError) {
      toast.error(message, toastConfig)
    } else {
      toast.success(message, toastConfig)
    }
  }

  render() {
    const repProviderValues = {
      repositories: this.state.repositories,
      originalRepositories: this.state.originalRepositories,
      updateRepositories: this.updateRepositories,
      refetchRepositories: this.refetchRepositories,
    }
    const toastProviderValues = { openToast: this.openToast }
    return (
      <RepositoryContext.Provider value={repProviderValues}>
        <ToastContext.Provider value={toastProviderValues}>
          <header className="bg-near-white fixed w-100 ph3 pv3 pv3-ns ph4-m ph5-l shadow-3">
            <nav className="flex items-center f7 fw6 tracked sans-serif">
              <span className="f5 link purple dib mr3" href="#" title="Home">GitHub Overview</span>
              <div className="dib h1 br mr3 o-20 b--mid-gray"></div>
              <Link className="ttu link dim mid-gray dib mr3" to="/">
                Home
              </Link>
              <Link className="ttu link dim mid-gray dib mr3" to="/repositories">
                Repositories
              </Link>
            </nav>
          </header>
          <div className="pt5">
            <ToastContainer />
            <Route exact path="/" component={HomePage} />
            <Route path="/repositories" component={RepositoriesPage} />
          </div>
        </ToastContext.Provider>
      </RepositoryContext.Provider>
    )
  }
}
