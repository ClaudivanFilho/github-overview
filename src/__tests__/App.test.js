import React from 'react'
import { mount } from 'enzyme'
import toJson from 'enzyme-to-json'

import App from '../components/App'

jest.mock('../api')

describe('Test App component', () => {
  let mountedComponent = null

  beforeAll(() => {
    mountedComponent = mount(
      <App />
    )
  })

  it('Renders correctly', () => {
    const wrapper = mount(<App />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('App should be defined', () => {
    expect(mountedComponent).toBeDefined()
  })

  it('Must find a Link component', () => {
    expect(mountedComponent.find('Link').exists()).toBe(true)
  })
})
