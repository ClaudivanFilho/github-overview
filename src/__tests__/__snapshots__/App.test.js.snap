// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App component Renders correctly 1`] = `
<App>
  <header
    className="bg-near-white fixed w-100 ph3 pv3 pv3-ns ph4-m ph5-l shadow-3"
  >
    <nav
      className="flex items-center f7 fw6 tracked sans-serif"
    >
      <span
        className="f5 link purple dib mr3"
        href="#"
        title="Home"
      >
        GitHub Overview
      </span>
      <div
        className="dib h1 br mr3 o-20 b--mid-gray"
      />
      <Link
        className="ttu link dim mid-gray dib mr3"
        to="/"
      >
        <a
          href="#"
        >
          Home
        </a>
      </Link>
      <Link
        className="ttu link dim mid-gray dib mr3"
        to="/repositories"
      >
        <a
          href="#"
        >
          Repositories
        </a>
      </Link>
    </nav>
  </header>
  <div
    className="pt5"
  >
    <ToastContainer
      autoClose={5000}
      bodyClassName={null}
      className={null}
      closeButton={
        <CloseButton
          ariaLabel="close"
        />
      }
      closeOnClick={true}
      draggable={true}
      draggablePercent={80}
      hideProgressBar={false}
      newestOnTop={false}
      pauseOnFocusLoss={true}
      pauseOnHover={true}
      position="top-right"
      progressClassName={null}
      rtl={false}
      style={null}
      toastClassName={null}
      transition={[Function]}
    >
      <div
        className="Toastify"
      />
    </ToastContainer>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    >
      <HomePage>
        <div
          className="w-100 flex flex-column"
        >
          <div
            className="w-100"
          >
            <h3
              className="mv2"
            >
              Your Pendencies
            </h3>
          </div>
        </div>
      </HomePage>
    </Route>
    <Route
      component={[Function]}
      path="/repositories"
    >
      <RepositoriesPageWithContext>
        <RepositoriesPage
          repositories={Array []}
        >
          <div
            className="flex flex-column w-100 pa3 sans-serif"
          >
            <div
              className="w-100 flex pb2"
            >
              <div
                className="w-33 flex"
              >
                <ComponentWithContext>
                  <AddRepository
                    openToast={[Function]}
                    refetchRepositories={[Function]}
                  >
                    <div
                      className="flex items-center"
                    >
                      <input
                        className="input-reset ba b--black-20 db w-auto pa0 pl2 bg-white br2"
                        id="repository-name"
                        onChange={[Function]}
                        placeholder="Owner/Name"
                        title="Add Repository"
                        value=""
                      />
                      <img
                        alt="add-icon"
                        className="flex items-center br7 ml2 pointer dim"
                        onClick={[Function]}
                        src="add.svg"
                        width="25"
                      />
                    </div>
                  </AddRepository>
                </ComponentWithContext>
              </div>
              <div
                className="w-33 flex justify-center"
              >
                <ComponentWithContext>
                  <Search
                    originalRepositories={Array []}
                    updateRepositories={[Function]}
                  >
                    <div
                      className="flex items-center"
                    >
                      <input
                        className="go-search-input input-reset ba b--black-20 db w-100 bg-white br2"
                        id="search"
                        onChange={[Function]}
                        placeholder="Search..."
                        type="text"
                        value=""
                      />
                      <img
                        alt="s-icon"
                        className="pl2 absolute"
                        src="glass.svg"
                        width="20"
                      />
                    </div>
                  </Search>
                </ComponentWithContext>
              </div>
              <div
                className="w-33 flex justify-end"
              >
                <ComponentWithContext>
                  <OrderBy
                    originalRepositories={Array []}
                    repositories={Array []}
                    updateRepositories={[Function]}
                  >
                    <div
                      className="flex items-center"
                    >
                      <select
                        className="pl2 bg-white br2 pointer"
                        id="orderBy"
                        onChange={[Function]}
                      >
                        <option
                          key="NAME_ASC"
                          value="NAME_ASC"
                        >
                          Name ASC
                        </option>
                        <option
                          key="NAME_DESC"
                          value="NAME_DESC"
                        >
                          Name DESC
                        </option>
                        <option
                          key="ISSUES_ASC"
                          value="ISSUES_ASC"
                        >
                          Issues ASC
                        </option>
                        <option
                          key="ISSUES_DESC"
                          value="ISSUES_DESC"
                        >
                          Issues DESC
                        </option>
                        <option
                          key="PRS_ASC"
                          value="PRS_ASC"
                        >
                          PR's ASC
                        </option>
                        <option
                          key="PRS_DESC"
                          value="PRS_DESC"
                        >
                          PR's DESC
                        </option>
                      </select>
                    </div>
                  </OrderBy>
                </ComponentWithContext>
              </div>
            </div>
            <div
              className="w-100"
            />
          </div>
        </RepositoriesPage>
      </RepositoriesPageWithContext>
    </Route>
  </div>
</App>
`;
